process/process.go

```diff
@@ -232,131 +180,71 @@ type Batcher interface {
 	switch cfg.Type {
 	case "aggregate":
-		var p procAggregate
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcAggregate(cfg)
 	case "aws_dynamodb":
-		var p procAWSDynamoDB
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcAWSDynamoDB(cfg)
 	case "aws_lambda":
-		var p procAWSLambda
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcAWSLambda(cfg)
 	case "base64":
-		var p procBase64
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcBase64(cfg)
 	case "capture":
-		var p procCapture
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcCapture(cfg)
 	case "case":
-		var p procCase
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcCase(cfg)
 	case "convert":
-		var p procConvert
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcConvert(cfg)
 	case "copy":
-		var p procCopy
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcCopy(cfg)
 	case "count":
-		var p procCount
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcCount(cfg)
 	case "delete":
-		var p procDelete
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcDelete(cfg)
 	case "dns":
-		var p procDNS
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcDNS(cfg)
 	case "domain":
-		var p procDomain
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcDomain(cfg)
 	case "drop":
-		var p procDrop
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcDrop(cfg)
 	case "expand":
-		var p procExpand
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcExpand(cfg)
 	case "flatten":
-		var p procFlatten
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcFlatten(cfg)
 	case "for_each":
-		var p procForEach
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcForEach(cfg)
 	case "group":
-		var p procGroup
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcGroup(cfg)
 	case "gzip":
-		var p procGzip
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcGzip(cfg)
 	case "hash":
-		var p procHash
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcHash(cfg)
 	case "http":
-		var p procHTTP
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcHTTP(cfg)
 	case "insert":
-		var p procInsert
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcInsert(cfg)
 	case "ip_database":
-		var p procIPDatabase
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcIPDatabase(cfg)
 	case "join":
-		var p procJoin
-		_ = config.Decode(cfg.Settings, &p)
-		return p, nil
+		return newProcJoin(cfg)
 	case "jq":
 		var p procJQ
 		_ = config.Decode(cfg.Settings, &p)
 		return p, nil
```

Comment:
refactor: missed JQ

Reply:
Great catch, bad rebases are the death of me. :)